docs/source/data_view.rst

```diff
@@ -29,6 +29,54 @@ When interpreting these values, the root row ``()`` corresponds to the
 The root row and column indices together refer to the cell in the top-left
 corner.
 
+Selections
+~~~~~~~~~~
+
+Implementers of the |IDataViewWidget| interface provide a |selection| trait
+that holds a list of tuples of selected row and column index values.  This
+trait is settable, so changes made to the trait are reflected in the selection
+in the view.
+
+The |selection_type| trait describes what gets selected when a user clicks
+on a cell.  It defaults to ``row``, which selects entire rows with one click,
+but implementations may optionally support ``item`` and ``column`` selection
+as well.
+
+In ``row`` selection type, the column values are all equal ``()`` (in other
+words, the indices of the appropriate row header), and users setting the
+values should adhere to that expectation.
+
+.. figure:: images/row_selection_type.png
+   :scale: 50
+   :alt: an illustration of row selection type
+
+   Row selection type.
+
+   This corresponds to the |selection| being set equal to
+   ``[((0,), ()), ((1, 0), ()), ((1, 2), ())]``.
+
+The ``column`` selection type only selects the column values that are children
+of a particular parent row, and so the row provided is that parent row.  Code
+which sets the value of the selection should adhere to that expectation.
+
+.. figure:: images/column_selection_type.png
+   :scale: 50
+   :alt: an illustration of column selection type
+
+   Column selection type.
+
+   This corresponds to the |selection| being set equal to
+   ``[((), (2,)), ((0,), (0,)), ((1,), (3,))]``.
+
+The |selection_mode| trait describes the behaviour of selections as the user
+interacts with them.  It defaults to ``extended``, which allows the user to
```

Comment:
This is because Qt has a distinctions between various multiple selection behaviours (see https://doc.qt.io/qt-5/qabstractitemview.html#SelectionMode-enum), and what we are using corresponds to "extended".  It is possible we might support other modes in the future (such as contiguous selection; wx is the limitation here and I am deliberately implementing just the minimum viable stuff) and I don't want to have confusion between different types of multiple selection.

Reply:
I see. Thank you.